plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -71,16 +71,29 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		return new WP_Error( 'no_plugins', __( 'No plugins found in the API response.', 'performance-lab' ) );
 	}
 
-	$plugins            = array();
-	$standalone_plugins = array_merge(
-		array_flip( perflab_get_standalone_plugins() ),
-		array( 'optimization-detective' => array() ) // TODO: Programmatically discover the plugin dependencies and add them here. See <https://github.com/WordPress/performance/issues/1616>.
-	);
-	foreach ( $response->plugins as $plugin_data ) {
-		if ( ! isset( $standalone_plugins[ $plugin_data['slug'] ] ) ) {
+	$plugins      = array();
+	$plugin_queue = perflab_get_standalone_plugins();
```

Comment:
I think the previous name of this variable, `$standalone_plugins`, makes more sense. It doesn't really need to be a queue, particularly related to my other comment that using a `while` loop seems unnecessarily complicated.

Reply:
I renamed it because in the `while` logic the dependancies are getting merged with the `$plugin_queue` so the `$standalone_plugins` variable name seems misleading as the dependancies are not standalone plugins.

Reply:
Makes sense.